src/recipe_dictionary.cpp

```diff
@@ -33,9 +33,57 @@ std::vector<const recipe *> recipe_subset::search( const std::string &txt ) cons
 {
     std::vector<const recipe *> res;
 
+    size_t split;
+    split = txt.find( ":" );
+    std::string key = "", value;
```

Comment:
In C++ std::string is always empty by default so you don't need the "" initialization. Also fundamental types can share a declaration but objects should be declared individually (so that if they later acquire parameters you don't need to do too much reformating).